modules/courses/public/views/pages/index.liquid

```diff
@@ -0,0 +1,35 @@
+---
+slug: course(/:id)
+layout: modules/courses/courses
+---
+{% liquid
+  function current_user = 'modules/user/lib/queries/user/current'
+
+  unless current_user.id
+    redirect_to '/user/register'
+  endunless
+
+  function can_manage_course = 'modules/permission/lib/helpers/can_do', requester: current_user, do: 'courses.manage'
+
+  assign status = '["published", "unpublished"]' | parse_json
+
+  graphql modules = 'modules/courses/module/search', withLessons: true, status: status
+  assign modules = modules.records.results
+
+  assign current_lesson = modules.first.lessons.first
+
+  if current_lesson == blank and can_manage_course
```

Comment:
what should happen if the current_lesson is blank but the user cannot manage the course? soft 404?

Reply:
> soft 404
yep. in our use case it shouldn't happen because the first thing we do is to add the course intro as an admin user, but ideally we should have a main coming soon page in case there is no content at all.
We'll add this (any many more nice features) to the standalone Courses module 👍